### PR TITLE
whisper: sym encryption message padding includes salt

### PR DESCRIPTION
**Pull request by gballet**
Now that the AES salt has been moved to the payload, padding must be adjusted to hide it, lest an attacker guesses that the packet uses symmetric encryption.

### CODE REVIEW

File: whisper/whisperv6/message.go [hunk 1]
```diff
@@ -124,6 +124,8 @@ func (msg *sentMessage) appendPadding(params *MessageParams) error {
 	rawSize := len(params.Payload) + 1
 	if params.Src != nil {
 		rawSize += signatureLength
+	} else {
+		rawSize += AESNonceLength
```
Comment (gluk256) [inline whisper/whisperv6/message.go]: this is true only for symmetric encryption

Comment (gballet) [reply whisper/whisperv6/message.go]: If `Src` is `nil`, then it's using symmetric encryption

Comment (gluk256) [reply whisper/whisperv6/message.go]: Src is used to identify the source of the message, i.e. for digital signature. Dst is used for asymmetrc encryption. And KeySym is used for symmetric encryption. So, you can change line 127 like this:
} else if params.KeySym != nil {

Comment (gballet) [reply whisper/whisperv6/message.go]: Well then it's very confusing. Then both sym and asym should be able to sign their messages. So it should be `if` and not `else if`

**Comment by gluk256** [comment]
:+1:

**Pull request merged**